R/telemetry.R

```diff
@@ -508,20 +514,54 @@ Telemetry <- R6::R6Class( # nolint object_name_linter
         checkmate::check_r6(session, "ShinySession", null.ok = TRUE),
         checkmate::check_class(session, "session_proxy")
       )
-
       self$data_storage$insert(
         app_name = self$app_name,
         type = type,
         session = session$token,
         details = details
       )
     },
-
+    .config_input_tracker = function(include_input_ids = NULL,
+                                     excluded_inputs = NULL,
+                                     excluded_inputs_regex = NULL,
+                                     track_values = FALSE) {
+      checkmate::assert_character(include_input_ids, null.ok = TRUE, any.missing = FALSE)
+      checkmate::assert_character(excluded_inputs, null.ok = TRUE, any.missing = FALSE)
+      checkmate::assert_character(excluded_inputs_regex, null.ok = TRUE, any.missing = FALSE)
+      checkmate::assert_logical(track_values, len = 1)
+      excluded_pattern <- ""
+      included_pattern <- ""
+      if (!is.null(excluded_inputs_regex)) {
+        excluded_pattern_regex <- paste(excluded_inputs_regex, collapse = "|")
+      } else {
+        excluded_pattern_regex <- ""
+      }
+      if (!is.null(excluded_inputs)) {
+        escaped_excludes <- sapply(excluded_inputs, function(input) {
```

Comment:
I'm not a huge fan of having to manipulate regex expressions like this.

How about adding a new return parameter for this `excluded_fixed` which would perform a `grepl(..., fixed = TRUE)` later on.

Or another way of flagging these comparisons as fixed.